catboost/private/libs/target/data_providers.cpp

```diff
@@ -606,16 +608,40 @@ namespace NCB {
 
         // Weights
         {
-            if (createClassTarget && !inputClassificationInfo.ClassWeights.empty()) {
+            if (createClassTarget && (!inputClassificationInfo.ClassWeights.empty() ||
+                inputClassificationInfo.AutoClassWeightsType != EAutoClassWeightsType::None))
+            {
+                auto targetClasses = !maybeConvertedTarget.empty() ?
```

Comment:
It could look better:
```
auto targetClasses = !maybeConvertedTarget.empty()
    ? TMaybe<TConstArrayRef<float>>(*maybeConvertedTarget[0])
    : Nothing();
```